src/CoreFoundation/CFProxySupport.cs

```diff
@@ -579,62 +581,241 @@ public static CFProxySettings GetSystemProxySettings ()
+		}
+
 		[DllImport (Constants.CFNetworkLibrary)]
 		extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
 			/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
-			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
-		
-		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString proxyAutoConfigurationScript, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
+			/* CFProxyAutoConfigurationResultCallback __nonnull */ CFProxyAutoConfigurationResultCallbackInternal cb,
+			/* CFStreamClientContext * __nonnull */ ref CFStreamClientContext  clientContext);
+
+		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (string proxyAutoConfigurationScript, Uri targetUrl, CFProxyAutoConfigurationResultCallback clientCb, CFStreamClientContext clientContext)
 		{
 			if (proxyAutoConfigurationScript == null)
-				throw new ArgumentNullException ("proxyAutoConfigurationScript");
+				throw new ArgumentNullException (nameof (proxyAutoConfigurationScript));
 
-			if (targetURL == null)
-				throw new ArgumentNullException ("targetURL");
+			if (targetUrl == null)
+				throw new ArgumentNullException (nameof (targetUrl));
 
-			if (resultCallback == null)
-				throw new ArgumentNullException ("resultCallback");
+			if (clientCb == null)
+				throw new ArgumentNullException (nameof (clientCb));
+			
+			using (var pacScript = new NSString (proxyAutoConfigurationScript)) 
+			using (var url = new NSUrl (targetUrl.AbsoluteUri)) {
+				if (url == null)
+					throw new ArgumentException ($"Could not get AbsoluteUri for {nameof (url)}");
 
-			if (clientContext == null)
-				throw new ArgumentNullException ("clientContext");
+				CreatePACCFRunLoopSource factory = delegate (CFProxyAutoConfigurationResultCallbackInternal cb, ref CFStreamClientContext context) {
+					return CFNetworkExecuteProxyAutoConfigurationScript (pacScript.Handle, url.Handle, cb, ref context);
+				};
+				return ExecutePacCFRunLoopSourceBlocking (factory, clientCb, clientContext);
+			}
+		}	
 
-			IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, resultCallback, clientContext);
-			return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+		public static CFProxy[] ExecuteProxyAutoConfigurationScript (string proxyAutoConfigurationScript, Uri targetUrl, out NSError outError)
+		{
+			outError = null;
+			IntPtr cbError = IntPtr.Zero;
+			if (proxyAutoConfigurationScript == null)
+				throw new ArgumentNullException (nameof (proxyAutoConfigurationScript));
+
+			if (targetUrl == null)
+				throw new ArgumentNullException (nameof (targetUrl));
+
+			using (var pacScript = new NSString (proxyAutoConfigurationScript)) 
```

Comment:
minor: generated bindings do

```csharp
var p = NSString.CreateNative (proxyAutoConfigurationScript);
try {
...
}
finally {
   NSString.ReleaseNative (p);
}
```

Reply:
@spouliot I'm using `using` since it is simpler and @rolfbjarne recommended to use it over manually releasing (diff instance, but was in the CFArray for the ParseProxies method).